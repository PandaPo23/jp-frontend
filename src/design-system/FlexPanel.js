import system from '../utils/System';
import Flex from './Flex';

const FlexPanel = system(
  'FlexPanel',
  {
    extend: Flex,
    flexDirection: 'column',
  },
  (props) => ({
    justifyContent: props.centered && 'center',
    alignItems: props.centered && 'center',
    minHeight: 0,
  })
);

export default FlexPanel;
