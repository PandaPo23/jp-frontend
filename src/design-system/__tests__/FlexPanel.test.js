import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FlexPanel from '../FlexPanel';

describe('FlexPanel component', () => {
  const getComponent = (props) => <FlexPanel />;
  it('should render as FlexPanel', () => {
    const component = getComponent();
    const wrapper = mount(component);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render centered FlexPanel', () => {
    const localProps = {
      centered: true,
    };
    const wrapper = mount(<FlexPanel {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
      wrapper
        .find(FlexPanel)
        .first()
        .prop('centered')
    ).toEqual(true);
  });
});
