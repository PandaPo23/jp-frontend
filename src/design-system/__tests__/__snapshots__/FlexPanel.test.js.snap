// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexPanel component should render as FlexPanel 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 0;
}

<FlexPanel
  display="flex"
  flexDirection="column"
>
  <div
    className="emotion-0"
  />
</FlexPanel>
`;

exports[`FlexPanel component should render centered FlexPanel 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 0;
}

<FlexPanel
  centered={true}
  display="flex"
  flexDirection="column"
>
  <div
    className="emotion-0"
  />
</FlexPanel>
`;
